src/content/docs/en/guides/integrations-guide/vercel.mdx

```diff
@@ -222,6 +222,74 @@ export default defineConfig({
 });
 ```
 
+### `isr`
+
+**Type:** <code>boolean | <a href="#vercelisrconfig">VercelISRConfig</a></code><br/>
+**Available for:** Serverless
+<Since v="7.2.0" pkg="@astrojs/vercel" />
+**Default**: `false`
+
+Allows your project to be deployed as an [ISR (Incremental Static Regeneration)](https://vercel.com/docs/incremental-static-regeneration) function, which caches your on-demand rendered pages in the same way as prerendered pages after first request. 
+
+To enable this feature, set `isr` to true in your Vercel adapter configuration in `astro.config.mjs`:
+
+```js title="astro.config.mjs" ins={8}
+import { defineConfig } from 'astro/config';
+import vercel from '@astrojs/vercel/serverless';
+
+export default defineConfig({
+  // ...
+  output: 'server',
+  adapter: vercel({
+    isr: true,
+  }),
+});
+```
+
+#### VercelISRConfig
+
+By default, an ISR function caches for the duration of your deployment. You can further control caching using `bypassToken`, `exclude`, and `expiration`.
+
+```js title="astro.config.mjs"
+import { defineConfig } from 'astro/config';
+import vercel from '@astrojs/vercel/serverless';
+
+export default defineConfig({
+  // ...
+  output: 'server',
+  adapter: vercel({
+    isr: {
+      /* 
+       * A secret random string that you create.
+       * 
+       * Its presence in the `__prerender_bypass` cookie will result in fresh responses being served, bypassing the cache. See Vercel’s documentation on [Draft Mode](https://vercel.com/docs/build-output-api/v3/features#draft-mode) for more information.
+       * Its presence in the `x-prerender-revalidate` header will result in a fresh response which will then be cached for all future requests to be used. See Vercel’s documentation on [On-Demand Incremental Static Regeneration (ISR)](https://vercel.com/docs/build-output-api/v3/features#on-demand-incremental-static-regeneration-isr) for more information.
+       * 
+       * @default `undefined`
+       */
+      bypassToken: process.env.BYPASS_TOKEN,
+      
+      /**
+       * Paths that will always be served by a serverless function instead of an ISR function.
+       * 
+       * @default `[]`
+       */
+    	exclude: [ "/api/invalidate" ],
+
+      /**
+       * Expiration time (in seconds) before the pages will be re-generated.
+       * 
+       * Setting to `false` means that the page will stay cached as long as the current deployment is in production.
+       * 
+       * @default `false`
+       */
+      expiration: 60 * 60 * 24;
+    },
+  }),
+});
+```
```

Comment:
``````suggestion
## Cache invalidation options

By default, ISR responses are cached for the duration of your deployment. You can further control caching by setting an `expiration` time or prevent caching entirely for certain routes.

### Time-based invalidation

You can change the length of time to cache routes this by configuring an `expiration` value in seconds:

```js
export default defineConfig({
    output: "server",
    adapter: vercel({
        isr: {
            // caches all pages on first request and saves for 1 day
            expiration: 60 * 60 * 24
        }
    })
})
```

### Manual invalidation

To implement Vercel's [Draft mode](https://vercel.com/docs/build-output-api/v3/features#draft-mode), or [On-Demand Incremental Static Regeneration (ISR)](https://vercel.com/docs/build-output-api/v3/features#on-demand-incremental-static-regeneration-isr), you can create a bypass token and provide it to the `isr` config along with the paths to exclude from caching: 

```js
export default defineConfig({
    output: "server",
    adapter: vercel({
        isr: {
            // A secret random string that you create.
            bypassToken: "005556d774a8",
            // Paths that will always be served fresh.
            exclude: [ "/api/invalidate" ] 
        }
    })
})
```
``````

Do not merge this! I just selfishly want to capture this as a PR comment before and after!  :smile:

Reply:
I'm struggling here a bit because I usually work by "mimicing" there is no equivalent section on this page. The way `VercelImageConfig` was documented (a single mention) ended up being misleading. Is this page supposed to be a reference, guide, or a tutorial?

Is there another integration guide I could look at to follow suite?

Reply:
I opened up to try some editing, and actually I think what fits this page is actually pretty much what I suggested above.

There's precedent for not defining / linking to `VercelImageConfig` (see `imagesConfig`) and as for what this page is, this is one page that has to do everything (as a result of us having inherited it from the package).

I think stating, and showing a code example for each thing someone might do separately is fine, and I think this can always be revisited to better split out reference from usage. Right now, what's helpful is seeing each of these ideas explained.

So, I'm pushing a commit based on what I just whipped up, and we can take it from there for this PR, and certainly look at restructuring our adapter docs in general down the line! @lilnasy